packages/grid/x-data-grid-premium/src/hooks/features/clipboard/useGridClipboardImport.ts

```diff
@@ -0,0 +1,404 @@
+  apiRef: React.MutableRefObject<GridPrivateApiPremium>,
+  props: Pick<
+    DataGridPremiumProcessedProps,
+    | 'pagination'
+    | 'paginationMode'
+    | 'processRowUpdate'
+    | 'onProcessRowUpdateError'
+    | 'getRowId'
+    | 'onClipboardPasteStart'
+    | 'onClipboardPasteEnd'
+    | 'experimentalFeatures'
+    | 'unstable_splitClipboardText'
+  >,
+): void => {
+  const processRowUpdate = props.processRowUpdate;
+  const onProcessRowUpdateError = props.onProcessRowUpdateError;
+  const getRowId = props.getRowId;
+  const enableClipboardPaste = props.experimentalFeatures?.clipboardPaste ?? false;
+  const rootEl = apiRef.current.rootElementRef?.current;
+
+  const splitClipboardText = props.unstable_splitClipboardText;
+
+  const handlePaste = React.useCallback<GridEventListener<'cellKeyDown'>>(
+    async (params, event) => {
+      if (!enableClipboardPaste) {
+        return;
+      }
+      if (!isPasteShortcut(event)) {
+        return;
+      }
+
+      const focusedCell = gridFocusCellSelector(apiRef);
+      if (focusedCell !== null) {
+        const cellMode = apiRef.current.getCellMode(focusedCell.id, focusedCell.field);
+        if (cellMode === 'edit') {
+          // Do not paste data when the cell is in edit mode
+          return;
+        }
+      }
+
+      if (!rootEl) {
+        return;
+      }
+
+      const text = await getTextFromClipboard(rootEl);
+      if (!text) {
+        return;
+      }
+
+      const pastedData = splitClipboardText(text);
+      if (!pastedData) {
+        return;
+      }
+
+      const cellUpdater = new CellValueUpdater({
+        apiRef,
+        processRowUpdate,
+        onProcessRowUpdateError,
+        getRowId,
+      });
+
+      const isSingleValuePasted = pastedData.length === 1 && pastedData[0].length === 1;
+
+      const cellSelectionModel = apiRef.current.unstable_getCellSelectionModel();
+      if (cellSelectionModel && apiRef.current.unstable_getSelectedCellsAsArray().length > 1) {
+        Object.keys(cellSelectionModel).forEach((rowId, rowIndex) => {
+          const rowDataArr = pastedData[isSingleValuePasted ? 0 : rowIndex];
+          const hasRowData = isSingleValuePasted ? true : rowDataArr !== undefined;
+          if (!hasRowData) {
+            return;
+          }
+          Object.keys(cellSelectionModel[rowId]).forEach((field, colIndex) => {
+            const cellValue = isSingleValuePasted ? rowDataArr[0] : rowDataArr[colIndex];
+            cellUpdater.updateCell({ rowId, field, pastedCellValue: cellValue });
+          });
+        });
+
+        cellUpdater.applyUpdates();
+        return;
+      }
+
+      const visibleColumnFields = gridVisibleColumnFieldsSelector(apiRef).filter((field) => {
+        if (columnFieldsToExcludeFromPaste.includes(field)) {
+          return false;
+        }
+        return true;
+      });
+
+      const selectedRows = apiRef.current.getSelectedRows();
+
+      if (selectedRows.size > 0 && !isSingleValuePasted) {
```

Comment:
This is not consistent with the priority table:
<img width="836" src="https://user-images.githubusercontent.com/13808724/236009386-0f897c1f-ba1c-40b0-bb4f-bdd1a4a000f0.png">

Current behavior: 
A single cell value is pasted into the selected cell (basically ignoring selected rows)

https://user-images.githubusercontent.com/13808724/236010232-8b6945cb-e3af-4d79-8685-8eebefd6cec3.mov

I think it's OK to stick to the current behavior and update the table.
What do you think? @joserodolfofreitas @MBilalShafi

Reply:
I have updated the priority table to reflect current behavior.

I have also extracted `defaultPasteResolver` function. It contains the resolution logic of pasting depending on the selection state. It's not overridable for now, but in the future, we might allow to override it to implement custom paste resolution if necessary.

As for copying to clipboard, I didn't think about it much, but maybe we can expose a resolver function for this instead of props like `disableCopyRowsToClipboard` and `disableCopyCellsToClipboard`, as I find them a bit confusing.

Reply:
Sounds good, exposing resolvers could be a nice customizability 👍

Reply:
> I think it's OK to stick to the current behavior.  

👍